komodo-test-utils/src/test/java/org/komodo/test/utils/AbstractLocalRepositoryTest.java

```diff
@@ -244,8 +249,46 @@ protected void useCustomCallback( final SynchronousCallback callback,
         }
     }
 
+    protected static UnitOfWork createTransaction( String user, String txName, boolean rollback, final UnitOfWorkListener callback ) throws Exception {
+        return _repo.createTransaction(user, txName , rollback, callback);
+    }
+
     private UnitOfWork createTransaction( final UnitOfWorkListener callback ) throws Exception {
-        return _repo.createTransaction( ( this.name.getMethodName() + '-' + this.txCount++ ), this.rollbackOnly, callback );
+        return createTransaction(TEST_USER, (this.name.getMethodName() + '-' + this.txCount++), this.rollbackOnly, callback);
+    }
+
+    protected static String txId(String... components) {
+        StringBuffer buf = new StringBuffer();
+
+        assertNotNull(components);
+        for (int i = 0; i < components.length; ++i) {
+            buf.append(components[i]);
+            if (i < (components.length - 1))
+                buf.append(HYPHEN);
+        }
+
+        return buf.toString();
+    }
+
+    /**
+     * @param user the owner of the transaction
+     * @return read-only transaction for the given user
+     * @throws Exception if error occurs
+     */
+    protected static UnitOfWork readTx(String user) throws Exception {
```

Comment:
Are the transactions created by `readTx`, `writeTx`, and `sysTx` ever committed or rollback'ed. I couldn't see where they were. But maybe I missed something.